### PR TITLE
Kubernetes: Add image list to clusters

### PR DESCRIPTION
**Pull request by bfarrell**
PR for cluster page for [#1047](https://github.com/spinnaker/spinnaker/issues/1047)

image

### CODE REVIEW

**Comment by bfarrell** [comment]
Added changes to project page as well.  The project changes are dependent on [clouddriver #869](https://github.com/spinnaker/clouddriver/pull/869)

**Comment by bfarrell** [comment]
@lwander PTAL

**Comment by lwander** [comment]
LGTM, WDYT @danielpeach?

File: app/scripts/modules/core/projects/dashboard/cluster/projectCluster.directive.html [hunk 1]
```diff
@@ -45,9 +45,12 @@
           </a>
         </td>
         <td>
-          <a class="heavy" ng-if="application.build.buildNumber" href="{{application.build.url}}" target="_blank">
+          <a class="heavy" ng-if="application.build.buildNumber && !application.build.images" href="{{application.build.url}}" target="_blank">
             <span>#</span>{{application.build.buildNumber}}
           </a>
+          <ul ng-if="application.build.images" style="list-style: none" ng-repeat="image in application.build.images">
```
Comment (danielpeach) [inline app/scripts/modules/core/projects/dashboard/cluster/projectCluster.directive.html]: We use inline styling in a bunch of places, but `class="list-unstyled"` feels like the better option here.

File: app/scripts/modules/core/serverGroup/serverGroup.directive.html [hunk 2]
```diff
@@ -17,12 +17,18 @@
...
-            <span class="server-group-sequence">{{ viewModel.serverGroupSequence }}</span><span ng-if="viewModel.jenkins">:</span>
+            <span class="server-group-sequence">{{ viewModel.serverGroupSequence }}</span><span ng-if="viewModel.jenkins || viewModel.images">:</span>
             <a ng-if="viewModel.jenkins"
                href="{{viewModel.jenkins.href}}"
                target="_blank">
               Build: #{{ viewModel.jenkins.number }}
             </a>
+            <span ng-if="viewModel.hasImages">
+              Images:
+              <ul style="list-style: none" ng-repeat="image in viewModel.images">
```
Comment (danielpeach) [inline app/scripts/modules/core/serverGroup/serverGroup.directive.html]: Here too.

File: app/scripts/modules/core/serverGroup/serverGroup.directive.js [hunk 3]
```diff
@@ -62,6 +63,8 @@ module.exports = angular.module('spinnaker.core.serverGroup.serverGroup.directiv
             if (serverGroup.buildInfo.buildInfoUrl) {
               viewModel.jenkins.href = serverGroup.buildInfo.buildInfoUrl;
             }
+          } else if (serverGroup.buildInfo && serverGroup.buildInfo.images) {
```
Comment (danielpeach) [inline app/scripts/modules/core/serverGroup/serverGroup.directive.js]: `_.get(serverGroup, 'buildInfo.images')` is cleaner

**Comment by danielpeach** [comment]
LGTM after comment

File: app/scripts/modules/core/serverGroup/serverGroup.directive.js [hunk 4]
```diff
@@ -42,6 +42,7 @@ module.exports = angular.module('spinnaker.core.serverGroup.serverGroup.directiv
             serverGroupSequence: $filter('serverGroupSequence')(serverGroup.name),
             jenkins: null,
             hasBuildInfo: !!serverGroup.buildInfo,
+            hasImages: !!serverGroup.buildInfo.images,
```
Comment (danielpeach) [inline app/scripts/modules/core/serverGroup/serverGroup.directive.js]: Why do a safe check of `serverGroup.buildInfo.images` below and not here?

Comment (bfarrell) [inline app/scripts/modules/core/serverGroup/serverGroup.directive.js]: Actually, this really isnt even needed.  Ill remove.

File: app/scripts/modules/core/projects/dashboard/cluster/projectCluster.directive.html [hunk 5]
```diff
@@ -45,9 +45,12 @@
           </a>
         </td>
         <td>
-          <a class="heavy" ng-if="application.build.buildNumber" href="{{application.build.url}}" target="_blank">
+          <a class="heavy" ng-if="application.build.buildNumber && !application.build.images" href="{{application.build.url}}" target="_blank">
             <span>#</span>{{application.build.buildNumber}}
           </a>
+          <ul class="list-unstyled" ng-if="application.build.images" ng-repeat="image in application.build.images">
```
Comment (danielpeach) [inline app/scripts/modules/core/projects/dashboard/cluster/projectCluster.directive.html]: One more thing -- shouldn't the ng-repeat be on the `<li>` insead of the `<ul>`?

Comment (bfarrell) [inline app/scripts/modules/core/projects/dashboard/cluster/projectCluster.directive.html]: Heh, you are correct.  My UI skills are showing.  This also gets rid of that extra line in there that was bothering me.  Ill do it for both.  Thanks!
image

**Comment by bfarrell** [comment]
@danielpeach @lwander Should be good now.  Let me know if you see anything else.

**Comment by danielpeach** [comment]
LGTM

**Comment by tomaslin** [comment]
I would just include image and tag and omit registry.

**Comment by bfarrell** [comment]
I had that originally, but I made a change based on @lwander's feedback in [clouddriver #855](https://github.com/spinnaker/clouddriver/pull/855) to use the KubernetesUtil which returns the registry as well.

I can change it back if thats the consensus but I think it makes sense in clouddriver as opposed to parsing out it back out here.

**Comment by lwander** [comment]
@bfarrell and I talked out of band - the fix to the registry prefix is in clouddriver so this is good to merge once it's rebased.

**Pull request merged**